### PR TITLE
Refactored s3upload.py to reduce imports

### PR DESCRIPTION
**Pull request by thimslugga**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Refactored some code in s3upload.py to reduce imports.

### CODE REVIEW

File: ec2rlcore/s3upload.py [hunk 1]
```diff
@@ -104,37 +86,30 @@ def get_presigned_url(uploader_url, filename, region="us-east-1"):
...
-    json_payload = json.dumps(put_dict)
+    payload = put_dict
 
     try:
-        response = requests.post(url=endpoint, data=json_payload)
-        # If the initial put was successful then proceed with uploading the file using the returned presigned URL
-        if response.status_code == 200:
+        response = requests.post(url=endpoint, json=payload)
+        # If the initial post was successful then proceed with uploading the file using the returned presigned URL
+        if(response.ok):
```
Comment (Drudenhaus) [inline ec2rlcore/s3upload.py]: The parentheses are redundant (a space works in their place).

File: ec2rlcore/s3upload.py [hunk 2]
```diff
@@ -161,7 +136,7 @@ def s3upload(presigned_url, filename):
         # The response puts the URL string in double quotes so cut off the first and last characters
         with open(filename, "rb") as payload:
             response = requests.put(url=presigned_url, data=payload)
-            if response.status_code == 200:
+            if(response.ok):
```
Comment (Drudenhaus) [inline ec2rlcore/s3upload.py]: The parentheses are redundant (a space works in their place).

File: ec2rlcore/s3upload.py [hunk 3]
```diff
@@ -104,37 +86,30 @@ def get_presigned_url(uploader_url, filename, region="us-east-1"):
...
-            the_request = urllib_request.Request(uploader_url)
-            uploader_url = urllib_urlopen(the_request).geturl()
-            query_str = urlparse.urlparse(uploader_url).query
-            put_dict = urlparse.parse_qs(query_str)
-    except urllib_urlerror:
-        pass
+        # Grabs the real URL if it has been shortened
+        uploader_url = requests.get(uploader_url).url
+        query_str = requests.utils.urlparse(uploader_url).query
+        put_dict = dict(x.split('=') for x in query_str.split('&'))
```
Comment (Drudenhaus) [inline ec2rlcore/s3upload.py]: This only works for valid S3 uploader URLs. Test test_main_upload_with_invalid_unparseable_support_url caught this error.

File: ec2rlcore/s3upload.py [hunk 4]
```diff
@@ -104,37 +86,30 @@ def get_presigned_url(uploader_url, filename, region="us-east-1"):
...
+    except requests.exceptions.Timeout:
+        raise requests.exceptions.Timeout("ERROR: Connection timed out.")
```
Comment (Drudenhaus) [inline ec2rlcore/s3upload.py]: It is redundant to catch and re-raise the same exception without adding any additional value.

File: ec2rlcore/s3upload.py [hunk 5]
```diff
@@ -104,37 +86,30 @@ def get_presigned_url(uploader_url, filename, region="us-east-1"):
...
     try:
-        if not put_dict:
-            the_request = urllib_request.Request(uploader_url)
-            uploader_url = urllib_urlopen(the_request).geturl()
-            query_str = urlparse.urlparse(uploader_url).query
-            put_dict = urlparse.parse_qs(query_str)
-    except urllib_urlerror:
-        pass
+        # Grabs the real URL if it has been shortened
+        uploader_url = requests.get(uploader_url).url
```
Comment (Drudenhaus) [inline ec2rlcore/s3upload.py]: This needs to be mocked in the tests.

File: ec2rlcore/s3upload.py (hunk 2 above)
Comment (thimslugga) [reply ec2rlcore/s3upload.py]: I'm aware, this is more a personal preference but I have no problem making the changes to meet the projects requirements.

File: ec2rlcore/s3upload.py (hunk 5 above)
Comment (thimslugga) [reply ec2rlcore/s3upload.py]: I might need some assistance with this as I've never dealt with writing tests / unit testing

File: ec2rlcore/s3upload.py (hunk 3 above)
Comment (thimslugga) [reply ec2rlcore/s3upload.py]: Let me take another look at this portion / test and I can refactor.

**Comment by gregbdunn** [comment]
It's been a few months with no changes - I'm going to go ahead and close this.

Feel free to open one up if you get around to those changes!

**Pull request closed without merging**